core/message_parsers.py

```diff
@@ -0,0 +1,350 @@
+        embeds = []
+        current_description = ""
+        page_number = 1
+
+        current_description += f"# {self.guild_name}\n\n"
+
+        for role in self.roles:
+            role_description = f"## {role.name}\n"
+            member_texts = []
+            for m in role.members:
+                rank_format = self._format_rank(m.rank)
+                member_text = f"{rank_format}{m.name}" if rank_format else m.name
+                member_texts.append(member_text)
+            
+            role_description += ", ".join(member_texts) + "\n\n"
+            
+            # Check if adding this role would exceed the limit
+            if len(current_description) + len(role_description) > 3000:
+                embeds.append(discord.Embed(description=current_description.strip(), colour=0x1ABC9C))
+                current_description = f"# {self.guild_name} (Continued)\n\n" + role_description
+                page_number += 1
+            else:
+                current_description += role_description
+
+        # Add statistics to the last embed
+        stats_description = (
+            f"## Guild Statistics\n"
+            f"**Total Members:** {self.total_members}\n"
+            f"**Online Members:** {self.online_members}\n"
+            f"**Offline Members:** {self.offline_members}\n"
+        )
+
+        embeds.append(discord.Embed(description=current_description.strip(), colour=0x1ABC9C))
+        current_description = f"# {self.guild_name} (Statistics)\n\n" + stats_description
+        page_number += 1
+
+        embeds.append(discord.Embed(description=current_description.strip(), colour=0x1ABC9C))
+
+        # Update titles with page numbers
+        total_pages = len(embeds)
+        for i, embed in enumerate(embeds, 1):
+            embed.title = f"{self.guild_name} - Page {i}/{total_pages}"
+
+        return embeds
+
+
+    def _format_online_embed(self) -> List[discord.Embed]:
+        return self._format_list_embed()
+
+    def _format_top_embed(self) -> List[discord.Embed]:
+        embed = discord.Embed(title=f"Guild Ranking List", colour=0x1ABC9C)
+        
+        description = []
+        description.append(f"# Top Guild Experience - {self.date.strftime('%m/%d/%Y')} (today)")
+        for entry in self.top_entries:
+            member = entry.member
+            rank_format = HypixelRank.format_rank(member.rank)
+            member_text = f"{rank_format}{member.name}" if rank_format else member.name
+            description.append(
+                f"### {entry.position}. {member_text}\n"
+                f"**{entry.experience:,}** Guild Experience"
+            )
+        
+        embed.description = "\n".join(description)
+        return [embed]
+
+    def _create_exp_graph(self, exp_data):
+        # Reverse data to show oldest to newest
+        dates, exp_values = zip(*reversed(exp_data))
+        
+        # Create figure with dark theme
+        plt.style.use('dark_background')
```

Comment:
Any chance we can use a transparent background with colors that work on both light & dark mode?

Reply:
hmmm, will try it

Reply:
also, done